cloudinit/config/cc_puppet.py

```diff
@@ -118,26 +119,17 @@ def __init__(
         self.csr_attributes_path = csr_attributes_path
 
 
-def _autostart_puppet(log):
+def _autostart_puppet(log, cloud: Cloud, service_name: str):
     # Set puppet to automatically start
-    if os.path.exists("/etc/default/puppet"):
-        subp.subp(
-            [
-                "sed",
-                "-i",
-                "-e",
-                "s/^START=.*/START=yes/",
-                "/etc/default/puppet",
-            ],
-            capture=False,
-        )
-    elif subp.which("systemctl"):
-        subp.subp(["systemctl", "enable", "puppet.service"], capture=False)
-    elif os.path.exists("/sbin/chkconfig"):
-        subp.subp(["/sbin/chkconfig", "puppet", "on"], capture=False)
-    else:
+    try:
+        if os.path.exists("/sbin/chkconfig"):
+            subp.subp(["/sbin/chkconfig", service_name, "on"], capture=False)
```

Comment:
how many systems out there still have `chkconfig`?

i think i would leave this out, if there's need support those systems, i would push that into `distro.manage_service`

Reply:
I had the same concern. I noticed that we could push it into chkconfig for mostly rhel systems. but if we have reason for it.

Reply:
I believe `chkconfig` was deprecated in favor of systemd's `systemctl`(systemd-based) or `service`(opensuse), `rcctl`(openbsd) or `rc-service`(alpine|gentoo) commands to manage services The original commit was > 10 years ago when that support was necessary to have a fallback in old sysvinit distros. I don't think we need this fallback anymore in tip of main and I don't want to pull that into cloudinit.distros if we can avoid it.  Let's actually drop this try/except handling on chkconfig as I don't think there are any images where want to support it anymore.   

Side note: When looking at 'old way' of doing things vs new way, we generally want to organize our code to prefer the new way and only fallback to the old way when new way doesn't work. Otherwise, we waste CPU cycles at boot for 99% or recent distributions unnecessarily checking for a file or behavior that no longer should exist.